modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/upgrade/v3_1_2/SXPBlueprintUpgradeProcess.java

```diff
@@ -0,0 +1,173 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.search.experiences.internal.upgrade.v3_1_2;
+
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONFactoryUtil;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+/**
+ * @author Felipe Lorenz
+ */
+public class SXPBlueprintUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		_upgradeSXPElement();
+
+		_upgradeSXPBlueprint();
+	}
+
+	private String _fixElementDefinition(String elementDefinition)
```

Comment:
If possible, I'd consider renaming it to something more descriptive. "fixXYZ" doesn't tell much.

Reply:
https://github.com/brianchandotcom/liferay-portal/commit/e871b34f75b1de96155b9ec9b3bf856a468e8f24

Reply:
@gustavolimav this is not about getting the PR accepted, but making readable and understandable code. For the others and for us, as well. Consider each and every upgrade process having a method `fixElement()` and you having to quickly figure out what each of these actually do. Compare then the same, when each of those methods are actually saying what they do, like `fixHelpTextLocalizations()`, `addDefaultTitles()` etc.

Reply:
Yeah, I agree 100% just put here as reference and so Felipe can change it in my code too